### ISSUE TITLE
Remove addons/strip-limits.libsonnet addon

### ISSUE THREAD

**Issue by paulfantom** [prometheus-operator/kube-prometheus#1500]
[README.md is still pointing to `addons/strip-limits.libsonnet` addon](https://github.com/prometheus-operator/kube-prometheus#stripping-container-resource-limits) as a way to strip containers from set resource limits. However since `release-0.8` resource requests are first-level settings and users should use the following form:
```
values+: {
  <component>+: {
    resources: {
      requests: { SET HERE },
      limits: { SET HERE },
    },
  },
}
```

README section should be removed and a new doc describing how to set resource requests and limits should be created. This also fits nicely with https://github.com/prometheus-operator/kube-prometheus/issues/1005

**Anything else we need to know?**:

Issue created as a result of https://github.com/prometheus-operator/kube-prometheus/pull/1498

**Comment by andrein** [comment]
@paulfantom I did this for all components in my cluster, however I can't find a way to set the resource requests for kube-rbac-proxy. 

![image](https://user-images.githubusercontent.com/52871/141187488-ac979475-02fc-4828-9d92-34236234e912.png)

**Comment by ArthurSens** [comment]
Good point @andrein, looking at the code it does seem like kube-rbac-proxy is kinda left out.

@paulfantom what do you think about kube-rbac-proxy being embedded into the `defaults` of the object that use them?

e.g. [node exporter](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/node-exporter.libsonnet) could have something like this:

```jsonnet
local krp = import './kube-rbac-proxy.libsonnet';

local defaults = {
  local defaults = self,
  // Convention: Top-level fields related to CRDs are public, other fields are hidden
  // If there is no CRD for the component, everything is hidden in defaults.
  name:: 'node-exporter',
  namespace:: error 'must provide namespace',
  version:: error 'must provide version',
  image:: error 'must provide version',
  resources:: {
    requests: { cpu: '102m', memory: '180Mi' },
    limits: { cpu: '250m', memory: '180Mi' },
  },
  .
  .
  .
  .
  kubeRbacProxy:: {
    name: 'kube-rbac-proxy'
    upstream: 'http://127.0.0.1:' + defaults.port + '/',
    secureListenAddress: '[$(IP)]:' + defaults.port,
    resources: {
      requests: { cpu: '100m', memory: '100Mi' },
      limits: { cpu: '150m', memory: '100Mi' },
    }
   .
   .
   .
  }
};

function(params) {
  local ne = self,
  _config:: defaults + params,

  local kubeRbacProxy = krp(ne._config.kubeRbacProxy),
  .
  .
  .
}
```

**Comment by andrein** [comment]
I like the idea!

On top of that, assuming that all the kube-rbac-proxies have roughly the same resource usage, I would also like to be able to specify those limits globally in [main.libsonnet](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/main.libsonnet) like so:

```
{
  values:: {
    common: {
      kubeRbacProxy: {
          resources:: {
            requests: { cpu: '102m', memory: '180Mi' },
            limits: { cpu: '250m', memory: '180Mi' },
        },
      },
    },
  },
}
```
or, maybe as a separate component like so:
```
{
  values:: {
    kubeRbacProxy: {
        resources:: {
          requests: { cpu: '102m', memory: '180Mi' },
          limits: { cpu: '250m', memory: '180Mi' },
      },
    },
  }
}
```

What do you think?

**Comment by paulfantom** [comment]
That would make sense. However I think this would clash with [the idea of reusing CRD-defined fields in `defaults` and reducing additional fields](https://github.com/prometheus-operator/kube-prometheus/pull/1477).

Ideally we would put the kube-rbac-proxy definition into `defaults.containers` list, but this is also problematic from UX pov.

I think for now we would need to keep this `strip-limits` addon, but reduce its scope only to additional containers.

BTW. why do you need to remove limits on kube-rbac-proxy container in the first place?
